### PR TITLE
feat: add eslint

### PR DESCRIPTION
**Pull request by lindsaylevine**
Fixes https://github.com/netlify/netlify-plugin-nextjs/issues/210

had to push --no-verify because it was hanging?? not sure why, checked/compared to `cli` to see if i was missing a dep or something 🤔

### CODE REVIEW

**Comment by lindsaylevine** [comment]
Screen Shot 2021-04-11 at 4 17 26 AM

**Comment by lindsaylevine** [comment]
also not sure about the ci failures with `run-s` 🤡

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -26,5 +26,8 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
+      - name: Linting
```
Comment (erezrokah) [inline .github/workflows/test.yml:29]: This should be after the `npm install` step to fix the `run-s` errors.
Also in CI actions it's better to use `npm ci` to have reproducible builds (also it's faster than `npm install`).

**Review by erezrokah — CHANGES_REQUESTED** [review]
> had to push --no-verify because it was hanging?? not sure why, checked/compared to cli to see if i was missing a dep or something 🤔

I thought it was hanging too, but then it reported over 2000 errors, so maybe it's just slow?

**Review by erezrokah — APPROVED** [review]

**Review by ehmicky — CHANGES_REQUESTED** [review]
:tada: 
Note: CI tests are failing.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const { overrides } = require('@netlify/eslint-config-node')
+
+const customOverrides = [
+  {
+    files: ['*.js'],
+    rules: {
```
Comment (ehmicky) [inline .eslintrc.js:6]: Note: instead of using overrides, it it possible to put those `rules` directly under the top-level `rules` property.

**Pull request closed without merging**